dotnet/src/SemanticKernel/Connectors/HuggingFace/TextEmbedding/HuggingFaceEmbeddingGeneration.cs

```diff
@@ -0,0 +1,119 @@
+        Verify.NotEmpty(model, "Model cannot be empty.");
+
+        this._model = model;
+
+        this._httpClient = new(httpClientHandler);
+
+        this._httpClient.BaseAddress = baseUri;
+        this._httpClient.DefaultRequestHeaders.Add("User-Agent", HttpUserAgent);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HuggingFaceEmbeddingGeneration"/> class.
+    /// Using default <see cref="HttpClientHandler"/> implementation.
+    /// </summary>
+    /// <param name="baseUri">Base URI for service API call.</param>
+    /// <param name="model">Model to use for service API call.</param>
+    public HuggingFaceEmbeddingGeneration(Uri baseUri, string model)
+    {
+        Verify.NotNull(baseUri, "Base URI cannot be null.");
+        Verify.NotEmpty(model, "Model cannot be empty.");
+
+        this._model = model;
+
+        this._httpClientHandler = new() { CheckCertificateRevocationList = true };
+        this._httpClient = new(this._httpClientHandler);
+
+        this._httpClient.BaseAddress = baseUri;
+        this._httpClient.DefaultRequestHeaders.Add("User-Agent", HttpUserAgent);
+    }
+
+    /// <inheritdoc/>
+    public async Task<IList<Embedding<float>>> GenerateEmbeddingsAsync(IList<string> data)
+    {
+        return await this.ExecuteEmbeddingRequestAsync(data);
+    }
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        this._httpClient.Dispose();
+        this._httpClientHandler?.Dispose();
+    }
+
+    #region private ================================================================================
+
+    /// <summary>
+    /// Performs HTTP request to given base URI for embedding generation.
+    /// </summary>
+    /// <param name="data">Data to embed.</param>
+    /// <returns>List of generated embeddings.</returns>
+    /// <exception cref="AIException">Exception when backend didn't respond with generated embeddings.</exception>
+    private async Task<IList<Embedding<float>>> ExecuteEmbeddingRequestAsync(IList<string> data)
+    {
+        try
+        {
+            var embeddingRequest = new EmbeddingRequest
+            {
+                Input = data
+            };
+
+            using var httpRequestMessage = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Post,
+                RequestUri = new Uri($"{EmbeddingEndpoint}/{this._model}", UriKind.Relative),
+                Content = new StringContent(JsonSerializer.Serialize(embeddingRequest)),
+            };
+
+            var response = await this._httpClient.SendAsync(httpRequestMessage).ConfigureAwait(false);
+            var body = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+
+            var embeddingResponse = JsonSerializer.Deserialize<EmbeddingResponse>(body);
+
+            return embeddingResponse?.Embeddings?.Select(l => new Embedding<float>(l.Embedding.ToArray())).ToList()!;
+        }
+        catch (Exception e) when (e is not AIException)
```

Comment:
recommend not catching critical exceptions as well
`catch (Exception e) when (!e.IsCriticalException())`

Reply:
Thanks, updated!